src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableDataSource.cs

```diff
@@ -67,5 +81,55 @@ public void Dispose()
         {
             tableManager.RemoveSource(this);
         }
+
+        public void Add(IAnalysisIssueVisualization hotspot)
+        {
+            var entry = new HotspotTableEntry(hotspot);
+            tableEntries.Add(entry);
+
+            lock (sinks)
+            {
+                foreach (var sink in sinks)
+                {
+                    sink.AddEntries(new[] {entry});
+                }
+            }
+
+            var hotspotFilePaths = hotspot.GetAllLocations().Select(x => x.CurrentFilePath);
+            IssuesChanged?.Invoke(this, new IssuesChangedEventArgs(hotspotFilePaths));
+        }
+
+        public event EventHandler<IssuesChangedEventArgs> IssuesChanged;
+
+        public IEnumerable<IAnalysisIssueLocationVisualization> GetLocations(string filePath)
+        {
+            var matchingLocations = tableEntries
+                .Select(entry => entry.Identity as IAnalysisIssueVisualization)
+                .SelectMany(hotspotViz => hotspotViz.GetAllLocations())
+                .Where(locationViz => PathHelper.IsMatchingPath(locationViz.CurrentFilePath, filePath));
+
+            return matchingLocations;
+        }
+
+        public void Refresh(IEnumerable<string> affectedFilePaths)
+        {
+            var changedEntries = tableEntries
+                .Where(entry => affectedFilePaths.Any(p =>
+                    PathHelper.IsMatchingPath(p, (entry.Identity as IAnalysisIssueVisualization).CurrentFilePath)))
```

Comment:
Using 'as' implies we think the object might not be an 'IAnalysisIssueVisualization'.
If we're certain of the type it would be clearer to use a hard cast.
If we're not certain of the type then we should use 'as' and handle the possibility that it could be null e.g. '?.CurrentFilePath' (assuming the PathHelper handles nulls...)